js/admin/aioseop-count-chars.js

```diff
@@ -3,69 +3,356 @@
  *
  * @since 2.9.2
  * @since 3.2.0 Moved to its own file.
+ * @since 3.3.0 Complete refactoring.
  */
 
+var pluginDirName;
+var currentPage;
 
-var aiosp_title_extra = parseInt( aioseop_count_chars.extraTitleLength, 10 );
+if ('undefined' !== typeof aioseopCharacterCounter) {
+	var aioseopCharacterCounterIsGutenberg = aioseopCharacterCounter.isGutenberg;
+	var aioseopCharacterCounterAutogenerateDescriptions = aioseopCharacterCounter.autogenerateDescriptions;
+	var extraTitleLength = +aioseopCharacterCounter.extraTitleLength;
+	currentPage = aioseopCharacterCounter.currentPage;
+	pluginDirName = aioseopCharacterCounter.pluginDirName;
+} else if ('undefined' !== typeof aioseopOGCharacterCounter) {
+	pluginDirName = aioseopOGCharacterCounter.pluginDirName;
+	currentPage = aioseopOGCharacterCounter.currentPage;
```

Comment:
Variable alignment with the `=` could make this easier to read, but the long variable name `aioseopCharacterCounterAutogenerateDescriptions` makes it rather difficult to.

Reply:
It does look better aligned so I've made the change. I'm going to change the variable from using `var` to `let` since I initially thought that Internet Explorer doesn't support this, but it looks like it only doesn't support it for `for` control structures.

Reply:
I've resorted to using `let` (ES6) to define them inside a block scope and shorten the variable names. It looks much better now, but as a result, `grunt-contrib-uglify` breaks because it doesn't support ES6. I've created an issue to migrate to TTerser instead (which is much popular at the moment and which does support ES6) - https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/3001. I'll make this change when I also tackle the issue to use minified JS files.